src/main/scala/is/hail/variant/HTSGenotypeView.scala

```diff
@@ -393,4 +393,91 @@ class HardCallStructView(rowType: TStruct, callField: String) extends HardCallVi
       throw new DataFormatException(s"Expected call to be non-negative, but found $gt")
     gt
   }
-}
\ No newline at end of file
+}
+
+class HardcallTrioGenotypeView(rs: TStruct, callField: String) {
+  private val trioTgs = rs.fieldType(3).asInstanceOf[TArray]
+  private val trioTg = trioTgs.elementType.asInstanceOf[TStruct]
+
+  require(trioTg.hasField("proband") && trioTg.hasField("mother") && trioTg.hasField("father"))
+  private val probandIndex = trioTg.fieldIdx("proband")
+  private val motherIndex = trioTg.fieldIdx("mother")
+  private val fatherIndex = trioTg.fieldIdx("father")
+
+  private val tg = trioTg.fieldType(probandIndex).fundamentalType.asInstanceOf[TStruct]
+
+  require(tg.hasField(callField))
+  private val gtIndex = tg.fieldIdx(callField)
+  require(tg.unify(trioTg.fieldType(motherIndex).fundamentalType) && tg.unify(trioTg.fieldType(fatherIndex).fundamentalType))
+
+  private var m: MemoryBuffer = _
+  private var gsOffset: Long = _
+  private var gsLength: Int = _
+  private var gOffset: Long = _
+  private var gProbandOffset: Long = _
+  private var gMotherOffset: Long = _
+  private var gFatherOffset: Long = _
+
+  var gIsDefined: Boolean = _
+
+  def setRegion(mb: MemoryBuffer, offset: Long) {
+    this.m = mb
+    gsOffset = rs.loadField(m, offset, 3)
+    gsLength = trioTgs.loadLength(m, gsOffset)
+  }
+
+  def setRegion(rv: RegionValue) {
+    setRegion(rv.region, rv.offset)
+  }
+
+  def setGenotype(idx: Int) {
+    require(idx >= 0 && idx < gsLength)
+    gIsDefined = trioTgs.isElementDefined(m, gsOffset, idx)
+    gOffset = trioTgs.loadElement(m, gsOffset, gsLength, idx)
+    gProbandOffset = trioTg.loadField(m, gOffset, probandIndex)
+    gMotherOffset = trioTg.loadField(m, gOffset, motherIndex)
+    gFatherOffset = trioTg.loadField(m, gOffset, fatherIndex)
```

Comment:
I don't think the `g` prefix is necessary here, it's clear that it's a genotype for the proband, etc.